homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,180 @@
+"""Support for Honeywell Lyric devices."""
+import logging
+from typing import Any, Dict
+
+import voluptuous as vol
+from lyric import Lyric
+
+from homeassistant.const import CONF_TOKEN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from .const import (
+    DATA_LYRIC_CLIENT,
+    DOMAIN,
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_LYRIC_CONFIG_FILE,
+    DATA_LYRIC_CONFIG,
+    SERVICE_HOLD_TIME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_CLIENT_ID): cv.string,
+                vol.Required(CONF_CLIENT_SECRET): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the Lyric component."""
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    # Store config to be used during entry setup
+    hass.data[DATA_LYRIC_CONFIG] = conf
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up Lyric from a config entry."""
+    conf = hass.data[DATA_LYRIC_CONFIG]
+
+    client_id = conf[CONF_CLIENT_ID]
+    client_secret = conf[CONF_CLIENT_SECRET]
+    token = entry.data[CONF_TOKEN]
+    token_cache_file = hass.config.path(CONF_LYRIC_CONFIG_FILE)
+
+    lyric = Lyric(
+        app_name="Home Assistant",
+        client_id=client_id,
+        client_secret=client_secret,
+        token=token,
+        token_cache_file=token_cache_file,
+    )
+
+    hass.data.setdefault(DOMAIN, {})[DATA_LYRIC_CLIENT] = LyricClient(lyric)
+
+    for component in "climate", "sensor":
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigType) -> bool:
+    """Unload Lyric config entry."""
+    for component in "climate", "sensor":
+        await hass.config_entries.async_forward_entry_unload(entry, component)
+
+    # Remove the climate service
+    hass.services.async_remove(DOMAIN, SERVICE_HOLD_TIME)
+
+    del hass.data[DOMAIN]
+
+    return True
+
+
+class LyricClient:
+    """Structure Lyric functions for hass."""
+
+    def __init__(self, lyric):
+        """Init Lyric devices."""
+        self.lyric = lyric
+
+        if not lyric.locations:
```

Comment:
It can happen now that `self._location` is never defined. Looks like you expect it to be an empty list in `dev devices`. I would remove this `if`. The list comprehension will then also result in empty list ?